Userland/Services/SystemServer/Service.h

```diff
@@ -96,7 +103,8 @@ class Service final : public Core::Object {
     // times where it has exited unsuccessfully and too quickly.
     int m_restart_attempts { 0 };
 
-    void setup_socket();
+    void setup_socket(unsigned socket_index);
```

Comment:
Let's make this take a `SocketDescriptor&` directly since `setup_socket` has no particular need to know the index of it in the vector.